integration-tests/src/test/resources/test-project/build.gradle.kts

```diff
@@ -0,0 +1,206 @@
+subprojects {
+
+    apply {
+        plugin("kotlin")
+        plugin("com.google.protobuf")
+        plugin("idea")
+        plugin("io.spine.mc-java")
+    }
+
+    dependencies {
+        api(Spine.base)
+
+        Protobuf.libs.forEach { implementation(it) }
+
+        // Force versions for compilation/runtime as well.
+        //
+        // Maybe, not all of them are required in this scope.
+        // This is to investigate later.
+        //
+        doForceVersions(configurations)
+    }
+
+    protobuf {
+        protoc {
+            artifact = Protobuf.compiler
+        }
+    }
+
+    // Apply a typical configuration to every module.
+    applyConfiguration()
+}
+
+/**
+ * The alias for typed extensions functions related to modules of this project.
+ */
+typealias Module = Project
+
+fun Module.applyConfiguration() {
+    configureJava()
+    configureKotlin()
+    setUpTests()
+    applyGeneratedDirectories()
+}
+
+fun Module.configureKotlin() {
+    kotlin {
+        explicitApi()
+        applyJvmToolchain(BuildSettings.javaVersion.asInt())
+    }
+
+    tasks.withType<KotlinCompile> {
+        setFreeCompilerArgs()
+        // https://stackoverflow.com/questions/38298695/gradle-disable-all-incremental-compilation-and-parallel-builds
+        incremental = false
+    }
+}
+
+fun Module.setUpTests() {
+    tasks.test {
+        useJUnitPlatform()
+
+        testLogging {
+            events = setOf(
+                TestLogEvent.PASSED,
+                TestLogEvent.FAILED,
+                TestLogEvent.SKIPPED
+            )
+            showExceptions = true
+            showCauses = true
+        }
+    }
+}
+
+fun Module.configureJava() {
+    java {
+        toolchain.languageVersion.set(BuildSettings.javaVersion)
+    }
+    tasks {
+        withType<JavaCompile>().configureEach {
+            configureJavac()
+            // https://stackoverflow.com/questions/38298695/gradle-disable-all-incremental-compilation-and-parallel-builds
+            options.isIncremental = false
+        }
+        withType<org.gradle.jvm.tasks.Jar>().configureEach {
+            duplicatesStrategy = DuplicatesStrategy.INCLUDE
+        }
+    }
+}
+
+/**
+ * Adds directories with the generated source code to source sets
+ * of the project and to IntelliJ IDEA module settings.
+ */
+fun Module.applyGeneratedDirectories() {
+
+    /* The name of the root directory with the generated code. */
+    val generatedDir = "${projectDir}/generated"
+
+    val generatedMain = "$generatedDir/main"
+    val generatedJava = "$generatedMain/java"
+    val generatedKotlin = "$generatedMain/kotlin"
+    val generatedGrpc = "$generatedMain/grpc"
+    val generatedSpine = "$generatedMain/spine"
+
+    val generatedTest = "$generatedDir/test"
+    val generatedTestJava = "$generatedTest/java"
+    val generatedTestKotlin = "$generatedTest/kotlin"
+    val generatedTestGrpc = "$generatedTest/grpc"
+    val generatedTestSpine = "$generatedTest/spine"
+
+    idea {
```

Comment:
I am not sure, whether we ever load this project as an IDEA module.

If we don't, this is a redundant stuff.